src/main/java/uk/gov/hmcts/reform/sendletter/entity/LetterRepository.java

```diff
@@ -94,4 +94,11 @@ int clearFileContent(
         + " WHERE l.id = :id AND l.status = 'Uploaded'"
     )
     int markStaleLetterAsCreated(@Param("id") UUID id, @Param("sentToPrintAt") LocalDateTime sentToPrintAt);
+
+    @Modifying(clearAutomatically = true)
+    @Query("UPDATE Letter l"
+        + " SET l.status = 'Aborted'"
+        + " WHERE l.id = :id"
```

Comment:
I suggest to add extra condition for current status of the letter similar to how it is done in `markStaleLetterAsNotSent` and `markStaleLetterAsCreated`, maybe

WHERE l.id = :id AND l.status <> 'Posted'

Reply:
Updated the where clause and added tests